src/flow/flow-test.ts

```diff
@@ -0,0 +1,162 @@
+import { Channel, Connection, ConsumeMessage } from "amqplib";
+import axios from 'axios';
+import { assert, expect } from 'chai';
+
+export enum IOType {
+    AMQP,
+    HTTP
+}
+
+export interface Input {
+    type: IOType;
+    payload: any;
+}
+
+export interface AmqpInput extends Input {
+    exchange: string;
+    routingKey: string;
+}
+
+export interface HttpInput extends Input {
+    url: string;
+}
+
+export interface Output {
+    type: IOType;
+    expectedOutput: any;
+}
+
+export interface AmqpOutput extends Output {
+    type: IOType;
+    queues: string[];
+    expectedQueue: string;
+}
+
+interface CreateMessageOut {
+    consumerPromise: Promise<void>;
+    messageCallback: (msg: ConsumeMessage | null) => any;
+}
+
+/**
+ * Returns a callback to be passed to channel.consume() and a promise that
+ * resolves or rejects when the callback processes a received message or the
+ * timeout is reached. Resolves when a message is received only when the message
+ * is expected to be received and the object equality assertion passes.
+ */
+function createReceiveMessageCallback(
+    expectedOutput: any,
+    outQueue: string,
+    shouldReceive: boolean
+): CreateMessageOut {
+    const res: any = {
+        resolve: undefined,
+        reject: undefined
+    };
+
+    const consumerPromise = new Promise<void>((resolve, reject) => {
+        res.resolve = resolve;
+        res.reject = reject;
+    });
+
+    const timeout = setTimeout(async () => {
+        if (shouldReceive) {
+            res.reject(new Error(`Queue ${outQueue} didn't receive a message before the timeout`));
+        } else {
+            res.resolve();
+        }
+    }, 6000);
+
+    return {
+        consumerPromise,
+        messageCallback: (msg: ConsumeMessage | null) => {
+            clearTimeout(timeout);
+
+            try {
+                if (msg === null) {
+                    throw new Error(`Output of queue ${outQueue} is null`);
+                }
+
+                if (!shouldReceive) {
+                    throw new Error(`Queue ${outQueue} received a message when it shouldn't have`);
+                }
+
+                const out = JSON.parse(msg.content.toString());
+
+                expect(out).to.deep.equal(expectedOutput, `Output of queue ${outQueue} does not match expected output`);
```

Comment:
``chai`` is a dev dependency.
You should either move it to prod dependency list or use an assertion library from [the standard library](https://nodejs.org/dist/latest-v12.x/docs/api/assert.html).